spec/validator/allowed_email_domain_validator_spec.rb

```diff
@@ -0,0 +1,54 @@
+require "rails_helper"
+
+class Validatable
+  include ActiveModel::Validations
+  attr_accessor :email
+
+  validates :email, allowed_email_domain: true
+end
+
+class ValidatableWithCurrentUser
+  include ActiveModel::Model
+  attr_accessor :current_user, :email
+
+  validates :email, allowed_email_domain: true
+end
+
+RSpec.describe AllowedEmailDomainValidator do
+  let(:model) { Validatable.new }
+
+  it "validates email with .gov.uk" do
+    model.email = "test.gov.uk"
```

Comment:
this doesn't look like a valid email address. is this correct?

Reply:
Good catch, I'll fix that